printing_auto_mrp/tests/test_printing_auto_mrp_production.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2025 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.exceptions import UserError
+from odoo.tools import mute_logger
+
+from odoo.addons.printing_auto_base.tests.common import (
+    TestPrintingAutoCommon,
+    patch_print_document,
+)
+
+
+@patch_print_document()
+class TestAutoPrinting(TestPrintingAutoCommon):
+    @classmethod
+    def setUpReportAndRecord(cls):
+        cls.report_ref = "mrp.action_report_production_order"
+        cls.record = cls.env.ref("mrp.mrp_production_4")
+
+    def setUp(self):
+        # Note: Using setUpClass, cls.record.picking_type_id.auto_printing_ids
+        # is reset on each test making them fail
+        super().setUp()
+        self.printing_auto = self._create_printing_auto_attachment()
+        self._create_attachment(self.record, self.data, "1")
+        self.record.picking_type_id.auto_printing_ids |= self.printing_auto
+
+    def test_button_mark_done_printing_auto(self):
+        self.printing_auto.printer_id = self.printer_1
+        self.record.button_mark_done()
+        self.assertFalse(self.record.printing_auto_error)
```

Comment:
IMO this is not ensuring that the printing happened. I would at least check that `handle_print_auto` is called.

Reply:
I've proposed a PR to improve tests: OCA/report-print-send#416